camkes-vm-linux: Update register new handler

ruizjuanpablo opened the pull request:
- reg_new_handler function was updated to vm_reg_new_vmcall_handler in
  the tutorial steps description.
- Rename vmm_vcpu_t to vm_vcpu_t in poke_handler
- Also updated the poke major device from 244 to 246
- Indicate that AddOverlayDirToRootfs is needed after the adding module
  instructions for the VM CMakeLists.txt
- Fixed some VM uppercase

Signed-off-by: Juan Pablo Ruiz juanpablo@ssrc.tii.ae

lsf37 commented on tutorials/camkes-vm-linux/camkes-vm-linux.md:
````diff
@@ -412,7 +412,7 @@ At the top of the file include our linux helpers, add the following:
 /*-- endfilter -*/
 ```
-Below the includes add:
+Below the includes (before `AddOverlayDirToRootfs` that was added the first exercise **Adding a program**) add:
````
```suggestion
Below the includes (before `AddOverlayDirToRootfs` that was added in the first exercise **Adding a program**) add:
```

lsf37 reviewed (approved):
Thanks for these! One small tweak, otherwise I think this can be merged as is.

ruizjuanpablo commented:
Hi @lsf37 , thanks for reviewing it. I have updated your change and added the sign-off-by to pass the DCO. I think you can now approve the running workflows now.
